pkg/cluster/cluster.go

```diff
@@ -496,3 +509,20 @@ func IsUnroutable(host string) bool {
 	}
 	return false
 }
+
+func makeConfig(networkType string) (*memberlist.Config, error) {
+	var mc *memberlist.Config
+
+	switch networkType {
+	case LanNetworkPeerType:
+		mc = memberlist.DefaultLANConfig()
+	case WanNetworkPeerType:
+		mc = memberlist.DefaultWANConfig()
+	case LocalNetworkPeerType:
+		mc = memberlist.DefaultLocalConfig()
+	default:
+		return mc, errors.New("unexpected network type, should be either: local, lan or wan")
```

Comment:
Maybe use the constants in the error message? `fmt.Errorf()` would achieve this in a concise way.

Grammar nit: 'either' refers to two options; 'should be one of' would be more accurate.

Reply:
actually `errors.Errorf` instead of `fmt.Errorf` but totally agree with @mattbostock here, we need to print `networkType` that was passed in error.

Minor nit: `return nil, errors....` instead of `mc` to be explicit that this is nil here.